### PR TITLE
[CI] Add missing step extract_branch

### PR DESCRIPTION
**Pull request by hcho3**
I messed up in #7382

### CODE REVIEW

File: .github/workflows/python_tests.yml [hunk 1]
```diff
@@ -148,9 +148,15 @@ jobs:
...
+      run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+      id: extract_branch
+      if: github.ref == 'refs/heads/master' || contains(github.ref, 'refs/heads/release_')
+
     - name: Upload Python wheel
       shell: bash -l {0}
-      if: (github.ref == 'refs/heads/master' || contains(github.ref, 'refs/heads/release_'))
+      if: github.ref == 'refs/heads/master' || contains(github.ref, 'refs/heads/release_')
       run: |
         python -m awscli s3 cp python-package/dist/*.whl s3://xgboost-nightly-builds/${{ steps.extract_branch.outputs.branch }}/ --acl public-read
```
Comment (hcho3) [inline .github/workflows/python_tests.yml:161]: This step will not work without first extracting the branch name from the `extract_branch` step.

**Review by trivialfis — APPROVED** [review]

**Comment by trivialfis** [comment]
Still doesn't show up on s3 .

**Comment by hcho3** [comment]
@trivialfis Here it is: https://s3-us-west-2.amazonaws.com/xgboost-nightly-builds/list.html?prefix=master/

**Comment by trivialfis** [comment]
Excellent!

**Comment by hcho3** [comment]
@trivialfis We need to backport the CI fixes back to `release_1.5.1` to get the MacOS artifact for XGBoost 1.5.1.

**Pull request merged**